File: src/main/java/me/captainpotatoaim/myplugin/listeners/Moving.java
package me.captainpotatoaim.myplugin.listeners;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;

public class Moving implements Listener {

    @EventHandler
    public void canMove(PlayerMoveEvent event) {
        if (!event.getPlayer().hasPermission("vix.move")) {
            event.setCancelled(true);
        }
    }

}

File: src/main/java/com/github/anviks/vixplugin/listeners/Moving.kt
package com.github.anviks.vixplugin.listeners

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent

class Moving : Listener {
    @EventHandler
    fun canMove(event: PlayerMoveEvent) {
        if (!event.getPlayer().hasPermission("vix.move")) {
            event.isCancelled = true
        }
    }
}
